tests/test_quart.py

```diff
@@ -0,0 +1,34 @@
+import msal
+import pytest
+from quart import Quart
+from identity.quart import Auth
+
+
+@pytest.mark.asyncio
+async def test_login(monkeypatch):
+    app = Quart(__name__)
+    app.config["SESSION_TYPE"] = "redis"
+    auth = Auth(
+        app,
+        authority="https://login.microsoftonline.com/123",
+        client_id="fake",
+        client_credential="fake",
+        redirect_uri="http://localhost:5000/auth_response",
+    )
+
+    monkeypatch.setattr(
+        msal.authority,
+        "tenant_discovery",
+        lambda *args, **kwargs: {
+            "authorization_endpoint": "https://login.microsoftonline.com/123/oauth2/v2.0/authorize",
+            "token_endpoint": "https://login.microsoftonline.com/123/oauth2/v2.0/token",
+        },
+    )
+
+    async with app.test_request_context("/", method="GET"):
+        rendered_template = await auth.login()
+        assert "Sign In" in rendered_template
+        assert (
+            "https://login.microsoftonline.com/123/oauth2/v2.0/authorize?client_id=fake&amp;response_type=code&amp;redirect_uri=http%3A%2F%2Flocalhost%3A5000%2Fauth_response&amp;scope=offline_access+openid+profile"
+            in rendered_template
+        )
```

Comment:
```suggestion
        assert (
            "https://login.microsoftonline.com/123/oauth2/v2.0/authorize"
            in rendered_template
        )
```
Will the test case still pass with this simplification?

Reason for removing "Sign In" is that it is a text which could easily be changed in the future.

Reason for removing the query parameter is that they are order-less so we aren't supposed to hard code their order. Besides, the correctness of the params are already guaranteed by the underlying auth library MSAL.

Reply:
Ah okay, sure, should pass, will change.